### ISSUE TITLE
Catalog templates cannot be reinstalled because of "Already Deployed"

### ISSUE THREAD

**Issue by dimw** [rancher/rancher#9942]
**Rancher versions:**
rancher/server: 1.6.7
rancher/agent: 1.6.7

**Infrastructure Stack versions:**
healthcheck: 0.3.1
ipsec: 0.1.3
network-services: 0.2.5
scheduler: 0.8.2
kubernetes (if applicable):

**Docker version: (`docker version`,`docker info` preferred)**
17.06.1-ce

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
Ubuntu 16.04.2 LTS (with KVM) (4.8.0)

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
Bare-metal

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
multi-node rancher, internal DB

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
1. Install stack using *DNSimple DNS* template (/catalog/community:infra*dnsimple) from the community catalog (using arbitrary credentials).
2. Remove the created *dnsimple* stack.
3. Try to reinstall the stack using the same template.

**Results:**
The stack cannot be reinstalled from the *dnsimple* template because it is marked as "Already Deployed" in the rancher UI even it is removed.
![2017-09-25_17-37-56](https://user-images.githubusercontent.com/2025688/30817173-612ac448-a218-11e7-8d1d-49e428a8e0fa.png)

**Comment by aemneina** [comment]
I believe if you have the dns simple docker/rancher compose files, you could in theory spin up another one under a different name. YMMV, you might want to try that.

**Comment by vincent99** [comment]
You probably just need to reload the page because the state of that button isn't updated every time a stack is removed...

**Comment by dimw** [comment]
@aemneina You are right, this probably would work. However, I'm interested in why the current situation has occurred and how a proper clean up can be performed.

@vincent99 Hahaha, good one! 👍 Seriously, I have already:
- reloaded the page couple of times (button is still disabled)
- tried to install the template using API (which complains about duplicate naming)
- reproduced the issue on another environment (installed, removed, and was not able to add it again)
- upgraded rancher/server and rancher/agent to v1.6.10.

In the meantime I found a container which is running on one of the hosts:
```
# docker ps -a | grep dnsimple
f272033e1c3f        rancher/external-dns:v0.6.0                                 "/.r/r /usr/bin/ex..."   2 minutes ago       Exited (1) 3 seconds ago                       r-dnsimple-dnsimple-1-dae6ea2c
```
It is restarting every ~15 seconds and ends with error exit code. Moreover, it is recreated (with a new container ID) even if it is removed manually.

It seems like the service was not removed from Rancher properly during the service removal process from the UI. Any ideas how to remove it manually from the Rancher DB? Maybe the removal does not work properly for services which are not able to start?

**Comment by vincent99** [comment]
It doesn't sound like the stack is removed, all the UI is looking for is a stack already using that catalog template ID.  Not being able to reuse the name also suggests the same.

**Comment by dimw** [comment]
@vincent99 Many thanks, it was a good hint! Basically, because _dnsimple_ is an infrastructure stack it was there but not visible from the default (User) stack list.

Indeed, by using the `rancher stack --system` command I was able to see that it is still present somewhere. Then (after months working with Rancher) I found out that there is a menu item in the UI which is _Stacks > All_ (which is not the same as clicking on _Stacks_ or selecting an environment) where I was able to find the _dnsimple_ stack. So after removal I was able to reinstall the stack again.
